### DISCUSSION TITLE
NETBOX issue: Server error - class 'django.core.exceptions.ImproperlyConfigured

### DISCUSSION THREAD

**Discussion by ld06000** [netbox-community/netbox#13138] [Q&A]
Hello all,

I have installed Netbox on a Ubuntu VM using the manual (I'm not a Linux man and not familiar with all LDAP, SQL, etc... world...)
It has been working fine until this afternoon . Now it sends:

![Netbox error](https://github.com/netbox-community/netbox/assets/131763602/697b979e-7816-4b93-9d42-50a7c0fc3b05)
Server Error
There was a problem with your request. Please contact an administrator.

The complete exception is provided below:

<class 'django.core.exceptions.ImproperlyConfigured'>

Required parameter AUTH_LDAP_SERVER_URI is missing from ldap_config.py.

Python version: 3.10.6
NetBox version: 3.5.4
If further assistance is required, please post to the [NetBox discussion forum](https://github.com/netbox-community/netbox/discussions) on GitHub.

I really don't know what is going on. Nothing ahs been changed on yhe VM nor the config...
Any idea ?
Many thank in advance for your help !!
Kind regards
Laurent

**Comment by ld06000** [comment]
Hello
Is there anyone able to help me ?
Thanks

**Reply by candlerb** [reply]
Not really. You have a very clear error message (*"Required parameter AUTH_LDAP_SERVER_URI is missing from ldap_config.py."*) but you haven't shown either configuration.py or ldap_config.py.

Netbox does not attempt to load ldap_config.py unless you explicitly [ask it](https://docs.netbox.dev/en/stable/installation/6-ldap/#configuration) to use the LDAPBackend for authentication.

You say you're "not familiar with all LDAP, SQL, etc"; but it seems that at some point, someone has attempted (partially) to configure Netbox to use LDAP authentication.

**Comment by ld06000** [comment]
is anyone able to help me?
I've been stuck by this for the last two weeks
thanks

**Reply by candlerb** [reply]
No, not unless you share your configuration.py and ldap_config.py as requested.

The error tells you there's something wrong in your config. That's where you need to look.

**Comment by ld06000** [comment]
ok thanks for you answer. Please find attached the two requested files in the zip file
many thanks in advance for your help!!![


[Netbox_config&ldap.zip](https://github.com/netbox-community/netbox/files/12159061/Netbox_config.ldap.zip)

**Reply by candlerb** [reply]
I see you have set this:

```
REMOTE_AUTH_BACKEND = 'netbox.authentication.LDAPBackend'
```

Even though you have REMOTE_AUTH_ENABLED set to False, I think you need to revert this to the [default](https://github.com/netbox-community/netbox/blob/v3.5.6/netbox/netbox/configuration_example.py#L195):

```
REMOTE_AUTH_BACKEND = 'netbox.authentication.RemoteUserBackend'
```

This is because setting remote auth to LDAPBackend forces Netbox to read ldap_config.py - and then complain when various mandatory settings are missing, as you are seeing.

EDIT: Reported as #13256

**Reply by ld06000** [reply]
Thank you so much!
I'm gonna try yhis and update afterward

**Reply by ld06000** [reply]
I've modified the configuration.py file and surprisingly it replies with a 504 error (time out)... The IP@ is the good one, I've restarted nginx... I'm lost...

**Reply by candlerb** [reply]
Comparing your configuration.py to the supplied example one, I see you've also set

```
        'SSL': True,
```

under `REDIS`, `'caching`'. That seems almost certain to be wrong: I suggest you put it back to `False`.

Apart from that, to rule out problems with nginx, try running Netbox in [development mode](https://docs.netbox.dev/en/stable/installation/3-netbox/#test-the-application).

I note that the configuration.py you supplied has Windows newlines (CR-LF), which might be OK, but makes me worry. What OS are you running Netbox under?

**Reply by ld06000** [reply]
thanks again Brian!
I'm running i on Ubuntu 22.4.2 LTS

**Comment by ld06000** [comment]
I went to the link https://docs.netbox.dev/en/stable/installation/3-netbox/#test-the-application and the following command is not recognized:
python3 manage.py  runserver 0.0.0.0:8000 --insecure
![python3_manage_py runserver](https://github.com/netbox-community/netbox/assets/131763602/f7b86466-ca7b-4e2b-bebb-86376f9ae2bc)

**Reply by kkthxbye-code** [reply]
Remove `--insecure` and you'll get the exception. When you figure out what the issue is and fix it, you can re-add the `--insecure` parameter again.

**Reply by candlerb** [reply]
You are probably running outside of the virtualenv (i.e. you didn't activate it).

Try this instead:

```
/opt/netbox/venv/bin/python /opt/netbox/netbox/manage.py runserver 0.0.0.0:8000 --insecure
```

**Reply by ld06000** [reply]
you're right! that worked much better like this
see screen capture attached
![manage_runserver](https://github.com/netbox-community/netbox/assets/131763602/332a848c-8e7c-43f0-8e7d-08e10124b2a3)

**Reply by candlerb** [reply]
Erm, you should have put '0.0.0.0:8000' not '0.0.0.0:0000'.

Ctrl-C to stop, run it again with `0.0.0.0:8000`, then point your web browser at `http://x.x.x.x:8000/` (where x.x.x.x is your server IP) and see if it works when you access it, or if it gives an error (look in the CLI where you're running manage.py for errors)

**Reply by ld06000** [reply]
now with 8000
![manage_runserver](https://github.com/netbox-community/netbox/assets/131763602/d5731932-211e-4d18-88a0-b67ae2108a0a)

**Reply by candlerb** [reply]
And what happens when you point your browser at `http://your.server.ip.address:8000/` ?

**Reply by ld06000** [reply]
This is what's happening when I point http://your.server.ip.address:8000/
 ![image](https://github.com/netbox-community/netbox/assets/131763602/a3693f3c-c124-4a5e-b9a9-8eb8b5792675)

**Reply by candlerb** [reply]
And that's happening while manage.py runserver is running? Then it seems you have some firewalling or other network issue.

**Comment by ld06000** [comment]
[manage.zip](https://github.com/netbox-community/netbox/files/12160109/manage.zip)
